src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.ValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private int size;
+    
+    @BeforeEach
+    void setUp() {
+        Storage.people.add(new User("Bob123", "111111111", 19));
+        Storage.people.add(new User("Alice123", "222222222", 22));
+        Storage.people.add(new User("John123", "333333333", 25));
+        size = Storage.people.size();
+    }
+    
+    @Test
+    void register_nonExistentLogin_Ok() {
+        User actual = registrationService.register(new User("Kate123", "111111111", 19));
+        assertEquals(size + 1, Storage.people.size());
+        assertNotNull(actual);
+    }

```

Comment:
better to use StorageDao interface
```suggestion
    @Test
    void register_successful() {
        User actual = registrationService.register(new User("Kate123", "111111111", 19));
        assertEquals(actual, storageDao.get(actual.getLogin()));
    }

```